### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by SergeBrakow**
a necessary restructuring of the code. So that triggered functionalities can work

### CODE REVIEW

**Review by Kheirah — CHANGES_REQUESTED** [review]

File: src/components/CustomerViewSection.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { ValueBox } from "./styles/ValueBox";
+import { MainBox } from "./styles/MainBox";
+
+export default function CustomerViewSection({customer, orderList}){
+    const navigate = useNavigate();
+    const ordersTheCustomerIsLinked = [];
+
+    {orderList.filter((order) =>
+        order.linkedCustomer.find((linkedCustomer) => linkedCustomer === customer.id) !== undefined? 
+            ordersTheCustomerIsLinked.push(order): ""
```
[the remark this answers is not in the document]
Comment (SergeBrakow) [reply src/components/CustomerViewSection.js]: fixed

File: src/components/CustomerViewSection.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                    {customer.contact_prson_note ===""?
+                        <P>-</P> :  <P>{customer.contact_prson_note}</P>
+                    }
+                </ValueBox>
+            </div>
+        </MainBox>
+    );
+}
+
+const Name = styled.p`
```
[the remark this answers is not in the document]
Comment (SergeBrakow) [reply src/components/CustomerViewSection.js]: fixed

File: src/components/NeedLoginMessage.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  return(
+  <div>
+    <StyledWelcomeName>
+      <h1>Cibri</h1>
+      <p>your service app</p>
+    </StyledWelcomeName>
+    <LoginDiv>
+      <fieldset>
+        <button onClick={()=> navigate("/")}>Sorry, but you need to login first!</button>
+      </fieldset>
```
[the remark this answers is not in the document]
Comment (SergeBrakow) [reply src/components/NeedLoginMessage.js]: fixed

File: src/components/UserViewSection.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                    <P>{user.name}</P>
+                    <P>Role: {user.role}</P>
+                </ValueBox>
+                <Name >Rechte</Name>
+                <ValueBox>
+                    {user.rights.map((right) => 
+                        <p key={right.id} >
+                            <input
+                                type="checkbox"
+                                checked={right.value === true ? true: false}
```
[the remark this answers is not in the document]
Comment (SergeBrakow) [reply src/components/UserViewSection.js]: fixed

File: src/pages/ErrorPage.js [hunk 5]
```diff
@@ -1,13 +1,84 @@
...
+import styled from "styled-components";
 
 export default function ErrorPage() {
   const navigate = useNavigate();
   return (
     <>
-      <h4>404 Page Not Found</h4>
-      <button onClick={() => navigate("/")}>to login</button>
+      <StyledWelcomeName>
+        <h1>Cibr2i</h1>
```
[the remark this answers is not in the document]
Comment (SergeBrakow) [reply src/pages/ErrorPage.js]: fixed

File: src/utils/localStorage.js [hunk 6]
```diff
@@ -20,10 +20,144 @@ export function fromLocalStorage(key){
...
     storage = [
-      { id: 1, name: "Peter", role:"user"}, 
-      { id: 2, name: "Nicole Schmidt", role: "admin"}
+      { 
+        id: "1",
+        name: "Tom",
+        role: "user",
+        rights : [ 
+          { id: 1,
+            name: "crateOrder",
```
[the remark this answers is not in the document]
Comment (SergeBrakow) [reply src/utils/localStorage.js]: done

File: src/components/UserLinkedInOrderSection.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+export default function UserLinkedInOrderSection({user, orderList}){
+    const navigate = useNavigate();
+    
+    const ordersTheUserIsLinkedIn = [];
+    {orderList.filter((order) =>
+        order.linkedUser.find((linkedUser) => linkedUser === user.id) !== undefined? 
+            ordersTheUserIsLinkedIn.push(order): ""
+        )
+    }
```
[the remark this answers is not in the document]
Comment (SergeBrakow) [reply src/components/UserLinkedInOrderSection.js]: fixed

**Pull request merged**